libbpf-cargo/src/test.rs

```diff
@@ -769,8 +769,94 @@ fn test_skeleton_builder_clang_opts() {
         .unwrap();
 }
 
-#[test]
-fn test_btf_dump_basic() {
+// -- TEST RUST GENERATION OF BTF PROGRAMS --
+
+/// Searches the Btf struct for a BtfType
+/// returns type identifier <u32> if found
+/// fails calling test if not found, or if duplicates
+///
+/// usage: -- search for basic struct/union/enum with exact match to name
+///        assert_type!(<Btf to search in>, <BtfType to search for>, <&str search name>);
+///        eg:
+///        let my_type = assert_type!(btf, Struct, "name")
+///    or: -- search for basic struct/union/enum/var containing name
+///        assert_type!(<Btf to search in>, <BtfType to search for>, <&str search name>, true);
+///        eg:
+///        let my_type = assert_type!(btf, Datasec, "bss", true);
+///    or: -- search for a Variable name in a Datasec -- exact match
+///        assert_type!(<Btf to search in>, Var, <&str search name>);)
+///        eg
+///        let let my_type = assert_type!(btf, Var, "name")
+macro_rules! assert_type {
```

Comment:
`assert_type` sounds misleading b/c `assert` family of macros don't return a value. Maybe this should be named `find_type_in_btf`? Implicit panicing in test helpers is reasonable to me -- it should be obvious in the event of a panic why the test failed.